app/controllers/ops_controller.rb

```diff
@@ -250,6 +208,76 @@ def change_tab(new_tab_id = nil)
 
   private ############################
 
+  def features
+    [{:role     => "ops_settings",
+      :name     => :settings,
+      :title    => N_("Settings")},
+
+     {:role     => "ops_rbac",
+      :role_any => true,
+      :name     => :rbac,
+      :title    => N_("Access Control")},
+
+     {:role     => "ops_diagnostics",
+      :name     => :diagnostics,
+      :title    => N_("Diagnostics")},
+
+     {:role     => "ops_analytics",
+      :name     => :analytics,
+      :title    => N_("Analytics")},
+
+     {:role     => "ops_db",
+      :name     => :vmdb,
+      :title    => N_("Database")},
+    ].map do |hsh|
+      ApplicationController::Feature.new_with_hash(hsh)
+    end
+  end
+
+  def set_active_elements(feature)
+    if feature
+      self.x_active_tree ||= feature.tree_list_name
+      self.x_active_accord ||= feature.accord_name
+    end
+    set_active_tab_and_node
+    get_node_info(x_node)
+  end
+
+  def set_active_tab_and_node
+    if x_active_tree == :settings_tree
+      @sb[:active_tab] ||= "settings_server"
+    end
```

Comment:
active_tab is to remember the tab user was on when they were last on that screen, i didn't want to touch that piece in this refactoring task.